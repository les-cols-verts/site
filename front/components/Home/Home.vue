<template>
  <main class="homepage">
    <picture>
      <source
        media="(min-width:650px)"
        srcset="../../static/agriculture_urbaine_xl.jpg"
      />
      <img
        src="../../static/agriculture_urbaine_xs.jpg"
        alt="Photo d'urbains dans un jardin"
        style="width: auto"
      />
    </picture>
    <section class="links">
      <app-link
        text="Qui sommes-nous ?"
        href="https://www.lescolsverts.com/collectifs/"
      ></app-link>
      <app-link
        text="Nos dispositifs d'agriculture urbaine"
        href="https://www.lescolsverts.com/services/"
      ></app-link>
      <app-link
        text="Découvrir nos actions nationales"
        href="https://www.lescolsverts.com/collectifs/"
      ></app-link>
      <app-link
        text="Nos actualités"
        href="https://www.lescolsverts.com/blog/"
      ></app-link>
      <app-link
        text="Nous Contacter"
        href="https://www.lescolsverts.com/contact/"
      ></app-link>
    </section>
  </main>
</template>

<style>
.homepage {
  margin-top: 20px;
}
</style>
<script>
import AppLink from '@/components/Link/AppLink';
export default {
  components: { AppLink },
};
</script>
<style scoped>
.homepage {
  display: flex;
  flex-direction: column;
  align-items: center;
}
.links {
  display: flex;
  flex-direction: column;
  max-width: 500px;
}
</style>
