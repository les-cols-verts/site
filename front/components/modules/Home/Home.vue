<template>
  <main class="homepage">
    <picture>
      <source
        media="(min-width:650px)"
        srcset="../../../static/agriculture_urbaine_xl.jpg"
      />
      <img
        src="../../../static/agriculture_urbaine_xs.jpg"
        alt="Photo d'urbains dans un jardin"
        style="width: auto"
      />
    </picture>
    <div class="map-cta">
      <HomeCta />
      <Map />
    </div>
  </main>
</template>

<style>
.homepage {
  margin-top: 20px;
}
</style>
<script>
import HomeCta from '@/components/modules/Home/HomeCta/HomeCta';
import Map from '@/components/modules/Home/Map/Map';

export default {
  components: { HomeCta, Map },
};
</script>
<style scoped>
.homepage {
  display: flex;
  flex-direction: column;
  align-items: center;
}
.map-cta {
  margin-bottom: 30px;
  margin-top: 30px;
}

@media only screen and (min-width: 1024px) {
  .map-cta {
    display: flex;
    flex-direction: row;
  }
}
</style>
