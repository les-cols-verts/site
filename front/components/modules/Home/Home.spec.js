import Home from '@/components/modules/Home/Home';

jest.mock('@/components/modules/Home/Map/Map', () => () => ({
  Map: '<div data-test="map"/>',
}));

describe('Home', () => {
  it('should match snapshot', () => {
    // When
    const wrapper = shallowMount(Home);

    // Then
    expect(wrapper).toMatchSnapshot();
  });
});
