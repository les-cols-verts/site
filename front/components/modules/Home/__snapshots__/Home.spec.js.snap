// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match snapshot 1`] = `
<main class="homepage">
  <picture>
    <source media="(min-width:650px)" srcset="../../../static/agriculture_urbaine_xl.jpg"> <img src="../../../static/agriculture_urbaine_xs.jpg" alt="Photo d'urbains dans un jardin" style="width: auto;">
  </picture>
  <div class="map-cta">
    <homecta-stub></homecta-stub>
    <!---->
  </div>
  <highlight-stub></highlight-stub>
</main>
`;
